lib/crawler.js

```diff
@@ -751,12 +938,11 @@ Crawler.prototype.queueURL = function(url, queueItem) {
     }
 
     // Pass this URL past fetch conditions to ensure the user thinks it's valid
-    var fetchDenied = false;
-    fetchDenied = crawler._fetchConditions.reduce(function(prev, callback) {
+    var fetchDenied = crawler._fetchConditions.reduce(function(prev, callback) {
         return prev || !callback(parsedURL);
     }, false);
 
-    if (fetchDenied) {
+    if (fetchDenied || crawler._robotsTxtFetchCondition(parsedURL)) {
```

Comment:
I don't really understand why there's a private function for this — I'm thinking we make it public and stick it directly on the prototype. Then if there's no robots.txt, or the checking has been disabled, we make that an initial check inside the body of the function and simply return true.